internal/controller/controller.go

```diff
@@ -507,18 +509,68 @@ func (c *Controller) deployConfiguration(config *dynamic.Configuration) error {
 		return fmt.Errorf("unable to find any active mesh pods to deploy config : %+v", config)
 	}
 
-	for _, pod := range podList.Items {
-		log.Debugf("Deploying to pod %s with IP %s", pod.Name, pod.Status.PodIP)
+	if err := deployToPods(podList.Items, config); err != nil {
+		return fmt.Errorf("error deploying configuration: %v", err)
+	}
 
-		if deployErr := c.deployToPod(pod.Name, pod.Status.PodIP, config); deployErr != nil {
-			return fmt.Errorf("error deploying configuration: %v", deployErr)
+	return nil
+}
+
+// deployConfigurationToUnreadyNodes deploys the configuration to the mesh pods.
+func (c *Controller) deployConfigurationToUnreadyNodes(config *dynamic.Configuration) error {
+	podList, err := c.clients.ListPodWithOptions(c.meshNamespace, metav1.ListOptions{
+		LabelSelector: "component==maesh-mesh",
+	})
+	if err != nil {
+		return fmt.Errorf("unable to retrieve pod list: %v", err)
+	}
+
+	if len(podList.Items) == 0 {
+		return fmt.Errorf("unable to find any active mesh pods to deploy config : %+v", config)
+	}
+
+	var unreadyPods []corev1.Pod
+
+	for _, pod := range podList.Items {
+		for _, status := range pod.Status.ContainerStatuses {
+			if !status.Ready {
+				unreadyPods = append(unreadyPods, pod)
+				break
+			}
 		}
 	}
 
+	if err := deployToPods(unreadyPods, config); err != nil {
+		return fmt.Errorf("error deploying configuration: %v", err)
+	}
+
 	return nil
 }
 
-func (c *Controller) deployToPod(name, ip string, config *dynamic.Configuration) error {
+func deployToPods(pods []corev1.Pod, config *dynamic.Configuration) error {
+	var errg errgroup.Group
+
+	for _, p := range pods {
+		pod := p
+
+		log.Debugf("Deploying to pod %s with IP %s", pod.Name, pod.Status.PodIP)
+
+		errg.Go(func() error {
+			b := backoff.NewExponentialBackOff()
+			b.MaxElapsedTime = time.Minute
```

Comment:
Did we want to fail after a minute? Perhaps 30 seconds might be a saner timeout.

Reply:
I'm ok with a quicker fail, since we have a 10s timer re-triggering a deploy to non-ready nodes

Reply:
quicker is better, I think as well 👍

Reply:
Sound good to me, will do !